<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Processing, plotting, and interactive eda of full-length scRNA-seq as SingleCellExperiments • chevreul</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Processing, plotting, and interactive eda of full-length scRNA-seq as SingleCellExperiments">
<meta name="description" content="Tools for managing SingleCellExperiment objects as projects. Includes functions for analysis and visualization of single-cell data. Also included is a shiny app for visualization of pre-processed scRNA data. Supported by NIH grants R01CA137124 and R01EY026661 to David Cobrinik.">
<meta property="og:description" content="Tools for managing SingleCellExperiment objects as projects. Includes functions for analysis and visualization of single-cell data. Also included is a shiny app for visualization of pre-processed scRNA data. Supported by NIH grants R01CA137124 and R01EY026661 to David Cobrinik.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">chevreul</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/cobriniklab/chevreul/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="chevreul">Chevreul<a class="anchor" aria-label="anchor" href="#chevreul"></a>
</h1></div>
<!-- badges: start -->

<p>This package includes a set of Shiny apps for interactive exploration of single cell RNA sequencing (scRNA-seq) datasets processed as <a href="https://bioconductor.org/packages/devel/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html" target="_blank" rel="noopener noreferrer" class="external-link">SingleCellExperiments</a></p>
<p>A demo with a developing human retina scRNA-seq dataset from Shayler et al. is available <a href="http://cobrinik-1.saban-chla.usc.edu:8080/app/objectApp" target="_blank" rel="noopener noreferrer" class="external-link">here</a></p>
<p>Chevreul includes tools for:</p>
<ul>
<li>Interactive data analysis and visualization</li>
<li>Louvain clustering at a range of resolutions</li>
<li>Dimensionality reduction of raw sequencing data.</li>
<li>Integration (batch correction) of multiple datasets <!--BB <(has been removed)-   Cell cycle state regression and labeling> -->
</li>
</ul>
<blockquote>
<p>[!WARNING] chevreul was designed for full-length smart-seq based single cell data. Default settings may not be appropriate for droplet (10x) data, though most can be adjusted. Keep in mind best practices regarding normalization, dimensional reduction, etc. when using.</p>
</blockquote>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Chevreul depends on a minimum R version 4.4. The current<br>
Chevreul loads three <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> packages</p>
<ol style="list-style-type: decimal">
<li><a href="https://github.com/cobriniklab/chevreulProcess" class="external-link"><code>chevreulProcess</code></a></li>
<li><a href="https://github.com/cobriniklab/chevreulPlot" class="external-link"><code>chevreulPlot</code></a></li>
<li><a href="https://github.com/cobriniklab/chevreulShiny" class="external-link"><code>chevreulShiny</code></a></li>
</ol>
<p>These enable standardized processing, plotting, and interactive analysis of SingleCellExperiments, respectively.</p>
<p><code>Chevreul</code> depends on a minimum R version &gt;=4.4 Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>. Then install <code>chevreul</code> and its dependencies using the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"cobriniklab/chevreul"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="troubleshooting-installation">Troubleshooting installation<a class="anchor" aria-label="anchor" href="#troubleshooting-installation"></a>
</h3>
<div class="section level4">
<h4 id="dependency-management">Dependency management<a class="anchor" aria-label="anchor" href="#dependency-management"></a>
</h4>
<p>When installing an R package like Chevreul with many dependencies, conflicts with existing installations can arise. This is a common issue in R package management. Here are some strategies to address this problem:</p>
<ol style="list-style-type: decimal">
<li><p>Consider <a href="https://rstudio.github.io/renv/articles/renv.html" target="_blank" rel="noopener noreferrer" class="external-link">renv</a> for dependency management. This tool creates isolated environments for each project, ensuring that package versions don’t conflict across different projects.</p></li>
<li><p>Use the conflicted Package The <a href="https://conflicted.r-lib.org" target="_blank" rel="noopener noreferrer" class="external-link">conflicted</a> package provides an alternative conflict resolution strategy. It makes every conflict an error, forcing you to choose which function to use</p></li>
</ol>
</div>
<div class="section level4">
<h4 id="slow-internet-connection">Slow internet connection<a class="anchor" aria-label="anchor" href="#slow-internet-connection"></a>
</h4>
<p>When installing R packages on slow internet connections, several issues can arise, particularly with larger packages or when using functions like remotes::install_github(). Here are some strategies to address bandwidth-related problems:</p>
<p>Set a longer timeout for downloads: <code>options(timeout = 9999999)</code></p>
<p>Specify the download method: <code>options(download.file.method = "libcurl")</code></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick Start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>After successful installation of the Chevreul package from Bioconductor, load the package along with the SingleCellExperiment example data provided with the package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cobriniklab/chevreul" class="external-link">chevreul</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load and view example dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tiny_sce"</span><span class="op">)</span></span>
<span><span class="va">tiny_sce</span></span></code></pre></div>
<div class="section level3">
<h3 id="run-clustering-on-a-single-object">Run clustering on a single object<a class="anchor" aria-label="anchor" href="#run-clustering-on-a-single-object"></a>
</h3>
<p>The <code>sce_clustering_workflow()</code> function performs key processing steps, including quality control filtering, normalization and log transformation, dimensionality reduction, Louvain clustering across various resolutions, and the identification of cluster-specific marker genes or transcripts.</p>
<!--  #BB default resolution is set to 0.6 ?  : By default clustering will be run at ten different resolutions between 0.2 and 2.0. Any resolution can be specified by providing the resolution argument as a numeric vector.-->
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tiny_sce_processed</span> <span class="op">&lt;-</span> <span class="fu">sce_clustering_workflow</span><span class="op">(</span><span class="va">tiny_sce</span>, resolution <span class="op">=</span> <span class="fl">0.6</span>, experiment_name <span class="op">=</span> <span class="st">"sce"</span>, organism <span class="op">=</span> <span class="st">"human"</span>, process <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="co">## BB: when Error in mat[, n_dimred, drop = FALSE] : subscript out of bounds</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-a-first-look-at-a-processed-dataset-using-an-interactive-shiny-app">Get a first look at a processed dataset using an interactive shiny app<a class="anchor" aria-label="anchor" href="#get-a-first-look-at-a-processed-dataset-using-an-interactive-shiny-app"></a>
</h3>
<p>The function <code>minimalChevreulApp</code> opens an interactive shiny application that can be used for visualization and exploration of the processed dataset.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">minimalChevreulApp</span><span class="op">(</span><span class="va">tiny_sce_processed</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="transcript-level-quantification">Transcript-level quantification<a class="anchor" aria-label="anchor" href="#transcript-level-quantification"></a>
</h2>
<p>For transcript-level analysis, users must incorporate transcript-level data into the SingleCellExperiment object as an alternative experiment before initiating the Chevreul processing pipeline. This step is crucial for enabling detailed exploration at the transcript level.</p>
<p>Transcripts may be quantified using any of several available methods, including alignment-free methods best used with well-annotated transcriptomes (Salmon, Kallisto), alignment-based methods best used to detect novel isoforms (StringTie2), or long-read methods for use with long-read sequencing data (IsoQuant).</p>
</div>
<div class="section level2">
<h2 id="integration-implementation">Integration implementation<a class="anchor" aria-label="anchor" href="#integration-implementation"></a>
</h2>
<p>The <code>sce_integrate()</code> function in Chevreul implements integration (batch correction) of scRNA-seq datasets by using the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/batchelor/inst/doc/correction.html" target="_blank" rel="noopener noreferrer" class="external-link">batchelor</a> package.</p>
<p>It accepts a list of SingleCellExperiment objects as input for integration and stores the corresponding batch information in a metadata field named ‘batch’. By default, it employs batchelor’s <code>correctExperiments</code> function to preserve pre-existing data structures and metadata from input SingleCellExperiment objects within the integrated output.</p>
</div>
<div class="section level2">
<h2 id="hardware-requirements">Hardware requirements<a class="anchor" aria-label="anchor" href="#hardware-requirements"></a>
</h2>
<p>Recommended minimum hardware requirements for running Chevreul are as follows:</p>
<ul>
<li>RAM: A minimum of 16 GB RAM is recommended for initial analysis. However, for larger datasets or more complex analyses, 64 GB or more is advisable.</li>
<li>CPU: Having multiple cores can be beneficial for parallel processing.</li>
<li>Storage: Sufficient storage space is necessary, especially for temporary files. The exact amount depends on the size of your datasets</li>
<li>R Version: Chevruel requires R version 4.4 or greater</li>
</ul>
<p>It’s important to note that these requirements can vary based on the size and complexity of your dataset. As the number of cells increases, so do the hardware requirements. For instance: A dataset with around 8,000 cells can be analyzed with 8 GB of RAM. For larger datasets or more complex analyses, 64-128 GB of RAM can be beneficial.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/cobriniklab/chevreul/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/cobriniklab/chevreul/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing chevreul</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Kevin Stachelek <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2085-695X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Bhavana Bhat <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/cobriniklab/chevreul/issues" class="external-link"><img src="https://img.shields.io/github/issues/cobriniklab/chevreul" alt="GitHub issues"></a></li>
<li><a href="https://github.com/cobriniklab/chevreul/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/cobriniklab/chevreul" alt="GitHub pulls"></a></li>
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://app.codecov.io/gh/cobriniklab/chevreul" class="external-link"><img src="https://codecov.io/gh/cobriniklab/chevreul/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kevin Stachelek, Bhavana Bhat.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
